script/openqa-clone-custom-git-refspec

```diff
@@ -160,6 +160,9 @@ Please try 'curl $json_url' or select another job, e.g. in the same scenario: $h
         local old_needledir
         old_needledir=$(echo "$json_data" | jq -r '.NEEDLES_DIR | select (.!=null)') || throw_json_error "$json_url" "$json_data"
         local needles_dir="${needles_dir:-"$old_needledir"}"
+        if [[ -n $needles_dir ]]; then
+            needles_dir=$(echo "$json_data" | jq -r '.NEEDLES_GIT_URL | select (.!=null)') || throw_json_error "$json_url" "$json_data"
+        fi
         needles_dir="${needles_dir:-"$old_productdir/needles"}"
```

Comment:
so if needles dir is not set then you set it but then in the following line there is still a check if it's not set and then sets it to `$old_productdir/needles`?

Reply:
i agree that it looks ugly. I just tried to inject a statement without change the final results. `$needles_dir` at that point is non-zero and overrides the `$old_needledir`

Reply:
`-n $needles_dir` is true if it is set.
https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html

Reply:
and it doesn't always set needles_dir, only if `NEEDLES_GIT_URL` is set.

Reply:
With this change the previous assignment `local needles_dir="${needles_dir:-"$old_needledir"}"` is never used.

I'm wondering how bad it would be to remove the whole `NEEDLES_DIR` handling here. What is actually the point of it? If the user didn't specify `NEEDLES_DIR` explicitly it makes most sense to just use the default.

Reply:
latest change https://github.com/os-autoinst/openQA/commit/0e4b252d2ddc25301b2f05164af20fcfcc61aaf0
move logic one step above and no check needles_dir anyway

Reply:
and latest VRs
https://openqa.opensuse.org/tests/4371052
https://openqa.opensuse.org/tests/4371035

Reply:
Can we close this?